pkg/providers/aws/provider_redis.go

```diff
@@ -681,6 +681,11 @@ func (p *RedisProvider) getElasticacheConfig(ctx context.Context, r *v1alpha1.Re
 		return nil, nil, nil, nil, errorUtil.Wrap(err, "failed to unmarshal aws elasticache cluster configuration")
 	}
 
+	// Override if node size is defined in the CR spec
+	if r.Spec.Size != "" {
+		elasticacheCreateConfig.CacheNodeType = aws.String(r.Spec.Size)
```

Comment:
Size (any string) -> CacheNodeType. Maybe need validation ?

Reply:
I would rather not maintain and perform validation to match the node sizes at https://aws.amazon.com/elasticache/pricing/ since these can change and are outside of our control

If you know an existing aws function that performs this validation, I can add it but otherwise, I think allowing the validation be done server side by aws itself is fine with me along as this error is propagated back up to the user in the CR status 🤔